mlrun/api/db/sqldb/db.py

```diff
@@ -1067,6 +1092,8 @@ def patch_feature_set(
                 f"Feature-set not found {feature_set_uri}"
             )
 
+        self._validate_feature_set_kind(feature_set_update.get("kind", "FeatureSet"))
```

Comment:
1. rename `feature_set_update` to `feature_set_patch`
2. do this validation only if `kind` is in the patch body